src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+const elements = [...document.querySelectorAll('li')];
+
+function getEmployees(employees) {
+  return employees.map((employee) => ({
+    name: employee.textContent.trim(),
+    position: employee.dataset.position,
+    salary: employee.dataset.salary,
+    age: employee.dataset.age,
+    element: employee,
+  }));
+}
+
+function sortList(list) {
+  const sortedEmployees = [...list].sort((employee1, employee2) => {
+    const salary1 = parseInt(employee1.salary.replace(/[^0-9]/g, ''), 10);
+    const salary2 = parseInt(employee2.salary.replace(/[^0-9]/g, ''), 10);
```

Comment:
The use of `replace(/[^0-9]/g, '')` is a good approach to extract numeric values from the salary string. Ensure that the salary data attribute is consistently formatted across all elements to avoid parsing errors.